lib/puppet/provider/package/pip3.rb

```diff
@@ -0,0 +1,22 @@
+# Puppet package provider for Python's `pip3` package management frontend.
+# <http://pip.openplans.org/>
+
+require 'puppet/provider/package/pip'
+
+Puppet::Type.type(:package).provide :pip3,
+  :parent => :pip do
+
+  desc "Python packages via `pip3`.
+
+  This provider supports the `install_options` attribute, which allows command-line flags to be passed to pip3.
+  These options should be specified as a string (e.g. '--flag'), a hash (e.g. {'--flag' => 'value'}),
+  or an array where each element is either a string or a hash."
+
+  def self.cmd
+    if Facter.value(:osfamily) == "RedHat" and Facter.value(:operatingsystemmajrelease).to_i < 7
+      "pip-python3"
```

Comment:
I'm not convinced there is a command called `pip-python3`. It could easily be that I just wasn't digging in the right places, but poking around on a centos 6 machine, there aren't any python3 packages readily available, but you can build from source which gives you `pip3`, not `pip-python3`. I know this logic is also in the pip provider, but I'm not convinced it's what we want here. Thoughts?